lib/WebService/Async/Segment/Customer.pm

```diff
@@ -0,0 +1,198 @@
+For a detailed information on the API call please refer to: L<https://segment.com/docs/spec/identify/>.
+
+It can be called with the following named params:
+
+=over
+
+=item * C<userId> - Unique identifier of a user (will overwrite object's attribute).
+
+=item * C<anonymousId> - A pseudo-unique substitute for a User ID (will overwrite object's attribute).
+
+=item * C<traits> - Free-form dictionary of traits of the user, like email or name (will overwrite object's attribute).
+
+=item * C<context> - Context information of the API call.
+Note that the API wrapper automatically sets context B<sentAt> and B<library> fields.
+
+=item * C<integrations> - A pseudo-unique substitute for a User ID, for cases when you don't have an absolutely unique identifier.
+
+=item * C<timestamp> - Dictionary of destinations to either enable or disable.
+
+=item * C<custom> - Dictionary of custom business specific fileds.
+
+=back
+
+About common fields please refer to: L<https://segment.com/docs/spec/common/>.
+
+=cut
+
+sub identify {
+    my ($self, %args) = @_;
+
+    my %call_args = $self->_make_call_args(\%args, [COMMON_FIELDS]);
+
+    return $self->api_client->method_call('identify', %call_args);
+}
+
+=head2 track
+
+Makes a B<track> call on the current customer. It can take any standard (B<event> and B<properties>), common or custom fields.
+For more information on track API please refer to L<https://segment.com/docs/spec/track/>.
+
+It can be called with the following parameters:
+
+=over
+
+=item * C<event> - required. event name.
+
+=item * C<properties> - Free-form dictionary of event properties.
+
+=item * C<userId> - Unique identifier of a user (will overwrite object's attribute).
+
+=item * C<anonymousId> - A pseudo-unique substitute for a User ID (will overwrite object's attribute).
+
+=item * C<traits> - Free-form dictionary of traits of the user, like email or name (will overwrite object's attribute).
+
+=item * C<context> - Context information of the API call.
+Note that the API wrapper automatically sets context B<sentAt> and B<library> fields.
+
+=item * C<integrations> - A pseudo-unique substitute for a User ID, for cases when you don't have an absolutely unique identifier.
+
+=item * C<timestamp> - Dictionary of destinations to either enable or disable.
+
+=item * C<custom> - Dictionary of custom business specific fileds.
+
+=back
+
+About common API call params: L<https://segment.com/docs/spec/common/>.
+
+=cut
+
+sub track {
+    my ($self, %args) = @_;
+
+    return Future->fail('Missing required argument "event"') unless $args{event};
+
+    my %call_args = $self->_make_call_args(\%args, [COMMON_FIELDS, qw(event properties)]);
+
+    return $self->api_client->method_call('track', %call_args);
+}
+
+sub _make_call_args {
+    my ($self, $args, $accepted_fields) = @_;
+
+    for (qw(userId anonymousId traits)) {
+        $args->{$_} ? ($self->{$_} = $args->{$_}) : ($args->{$_} = $self->{$_});
+    }
```

Comment:
why is `track` method  mutating userId in this object? maybe better to just take these fields from object.

Reply:
I implemented it this way for flexibility. There are imaginable scenarios for events that update ids and traits; for example when a client creates an account after anonymously browsing the website, or when a client makes changes to their profile. With current implementation we can update customer object by any event that updates profile information.

Reply:
Taking a second look at it, I think I've got to agree with you. Even in the mentioned scenarios, updating customer info doesn't take effect, unless `identify` is separately called.  So it doesn't make sense to do any automatic update on any api call, except `identify` call itself.